### ISSUE TITLE
Support more values for replacement 

### ISSUE THREAD

**Issue by ekeren** [rollout/rollout.io-ios#28]
Currently only primitives, NSString and NSData are available for replacement. 

UITableViewCell
NSNumber
NSArray
NSDictionary 

Please feel free to add more types in comments  here

**Comment by sbhhbs** [comment]
NSDate +1

**Comment by glencoates** [comment]
Would love to have this for anything where you can express the values using only built-in classnames and primitives, e.g.

```
@[@"a", @"b"]
@{@"a": @"b"}
@2
[NSDecimalNumber decimalNumberWithString:@"12.2"]
```

Would it be possible to run text input through an objc parser and determine that it can execute without any dependencies on the surrounding execution context and if so, accept it as a new return value?

**Comment by ekeren** [comment]
@glencoates - can you explain the use case a bit?

**Comment by glencoates** [comment]
Hi Eyal - happy to.

The specific issue that caused me to vote for this was that we had a static method that returned a key-value mapping of field names to labels as an NSDictionary.  We had a bug in that method so that it returned the wrong thing, and since those logical field names would sometimes actually need to map directly to CoreData field names, this would cause a bug.

What we would have loved to be able to do would be swizzle that method to return a fixed NSDictionary of the correct field name mapping where all the keys and values were NSStrings.

As I thought about this, it seemed like if you were going to allow Rollout to accept an arbitrary NSDictionary as a replacement return value, why not allow it to accept any valid Objective-C expression that returns an NSObject and can be successfully evaluated by the runtime without any additional input / context?

In other words, if you would accept `@{@"a": @"b"}` as a replacement value, why not `[NSDictionary dictionaryWithObjectsAndKeys: @"b", @"a", nil]`, and if you'd accept that, why not `[[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:nil]`?

Hope that helps understand what I was thinking?

**Comment by ekeren** [comment]
All objc types can be created via javascript see how this [example](http://support.rollout.io/docs/javascript-examples#create-a-new-instance)
